File: frontend/src/api/generation.ts
import apiClient from './client';
import type { Generation, GenerationCreateRequest } from '@/types';

export const generationApi = {
  create: async (request: GenerationCreateRequest): Promise<Generation> => {
    const { data } = await apiClient.post('/api/v1/generation', request);
    return data;
  },

  getById: async (id: string): Promise<Generation> => {
    const { data } = await apiClient.get(`/api/v1/generation/${id}`);
    return data;
  },

  getStatus: async (id: string): Promise<{
    id: string;
    status: string;
    result_url: string | null;
    result_file_url: string | null;
    error_message: string | null;
  }> => {
    const { data } = await apiClient.get(`/api/v1/generation/${id}/status`);
    return data;
  },

  getAll: async (
    offset = 0,
    limit = 50,
    generationType?: string
  ): Promise<{ items: Generation[]; total: number }> => {
    const { data } = await apiClient.get('/api/v1/generation', {
      params: {
        offset,
        limit,
        ...(generationType && { generation_type: generationType }),
      },
    });
    return data;
  },
};

File: src/api/schemas/generation.py
"""Generation schemas."""

from datetime import datetime
from uuid import UUID
from pydantic import BaseModel
from src.shared.enums import GenerationStatus, GenerationType


class GenerationCreateRequest(BaseModel):
    """Create generation request."""
    model_code: str
    prompt: str | None = None
    image_url: str | None = None
    video_url: str | None = None  # NEW: для motion control
    aspect_ratio: str = "1:1"
    duration: int | None = None
    extra_params: dict | None = None


class GenerationResponse(BaseModel):
    """Generation response."""
    id: UUID
    generation_type: GenerationType
    status: GenerationStatus
    prompt: str | None
    tokens_spent: int
    result_url: str | None
    result_file_url: str | None = None
    error_message: str | None
    created_at: datetime
    completed_at: datetime | None
    model_code: str | None = None
    model_name: str | None = None

    class Config:
        from_attributes = True


class GenerationListResponse(BaseModel):
    """List of generations response."""
    items: list[GenerationResponse]
    total: int


class GenerationStatusResponse(BaseModel):
    """Generation status response."""
    id: UUID
    status: GenerationStatus
    result_url: str | None = None
    result_file_url: str | None = None
    error_message: str | None = None
    progress: int | None = None

